src/soar_sdk/code_renderers/action_renderer.py

```diff
@@ -258,7 +268,9 @@ def render_outputs_ast(
                     )
             else:
                 keywords = []
-                if (extras := {**field.field_info.extra}) or field_name.modified:
+                extras = {**parse_json_schema_extra(field.json_schema_extra)}
```

Comment:
No need to unpack here, just take the whole dict.

```suggestion
                extras = parse_json_schema_extra(field.json_schema_extra)
```

Reply:
If we need to copy this, let's explicitly `.copy()` it. Might make sense to do that inside the helper func.